### DISCUSSION TITLE
random network request fails

### DISCUSSION THREAD

**Discussion by diffent** [supabase/supabase#33181] [Troubleshooting]
# Bug report

## Describe the bug

Random fails from react-native Supabase queries of these forms:

"TypeError: Network request failed"
"TypeError: Network request timed out"

(then a stack trace)

## To Reproduce

It is random but happening a lot today.

## Expected behavior

Supabase calls should succeed.

## Screenshots

N/A

## System information

iOS expo react-native 

## Additional context

Sometimes supabase queries work, sometimes the same queries fails.
What is going on?

**Comment by ahanusek** [comment]
I observe hundreds of errors like this daily. Supabase Support claims that it is a network issue, but the problem is that other requests work fine at the same time. This clearly points to an issue with the Supabase SDK or Supabase Database. However, I have no idea how to find any relevant information in the logs. Today, I'm also seeing thousands of DB connection timeout errors for my backend service connecting to Supabase DB (via Kysely). In the pooler, the only information I can find is:

"ClientHandler: socket closed with reason {:badmatch, :error}"

Compute size: Large

<img width="247" alt="Zrzut ekranu 2024-10-16 o 19 01 44" src="https://github.com/user-attachments/assets/dbc518bc-ec99-4cb1-bc69-7e860d6c860d">
<img width="916" alt="Zrzut ekranu 2024-10-16 o 19 01 20" src="https://github.com/user-attachments/assets/ffcbe2fd-3489-432c-aeaf-5291efbcb845">

**Comment by diffent** [comment]
That's for the added info here.

**Comment by diffent** [comment]
hello, we found that the network timeouts on supabase calls are due to client side not server side (so far as we can tell).  what was happening is that our application was making too many supabase calls within a short time.  and the client device can only process 12 or 16 or something along those lines at one time even if you have a fast network conn.  so the remainder get queued up.  eventually the later queued ones hit the 60 second or so default timeout.  we were able to track this by intercepting the supabase fetch calls (this is in javascript api of supabase).  code snippet below.  look for function ourFetch below.

partial solution is to batch calls (e.g. instead of trying to pull 1 row at a time from a table, pull a whole bunch of rows at once).  other issues we see are repeated calls to the same supabase endpoint due to "reactive" type code.  reactive = too many renders, too many data queries.  :)  that is a client side code problem we have to fix.

```
let fetchCount = 0
let launchTime = 0

if (!supabaseUrl || !supabaseAnonKey) {
  throw new Error('Supabase URL and/or Anon Key are missing.');
}

const ourFetch = (...args) => {
  //console.log("in our fetch", JSON.stringify(args, null, 2))
  if (!launchTime) launchTime = Date.now()
  console.log("url", args[0])
  console.log("fetchCount ms", fetchCount++, Date.now()-launchTime)
  return fetch(...args)
}

export const supabase = createClient(supabaseUrl, supabaseAnonKey, {
  auth: {
    storage: AsyncStorage,
    autoRefreshToken: true,
    persistSession: true,
    detectSessionInUrl: false,
  },
  realtime: {
    params: {
      eventsPerSecond: 1,
    },
  },
  global: {
    fetch: ourFetch
  }
});
```

**Comment by marius-se** [comment]
I'm also experiencing this - **even when running Supabase locally**. In my case it's definitely related to heavy loads/ lots of requests hitting Supabase at the same time. I have a migration script that pushes a lot of data to Supabase in a short amount of time, throughout hundreds/ thousands of requests. I'll try to batch the calls now.

**Comment by marius-se** [comment]
Also found this post: https://www.reddit.com/r/Supabase/comments/1g2okq4/intermittent_error_typeerror_fetch_failed/
Pretty much the same

**Comment by diffent** [comment]
batching definitely fixed it for us.  can't spam the Supabase API with lots of tiny calls....

**Comment by govi218** [comment]
I have this same issue, I've profiled my app to ensure minimal requests are made/no rerender spam to the servers yet I still get these. It's quite hard to repro which makes it all the more frustrating.

**Comment by justinschier** [comment]
I am having the same issue. I'll have several identical calls that take < 100ms, but some randomly take 20 seconds or time out completely! Hard for me to believe I'm one of the very few people seeing this. I see there are current capacity issues on the Supabase status page, but my region, us-east-1 is not mentioned at all.

**Comment by govi218** [comment]
<img width="798" alt="Screenshot 2025-01-12 at 00 38 23" src="https://github.com/user-attachments/assets/f42cc787-73c3-4aa7-a078-1a472f59f0b5" />

I talked to Supabase support and this is their recommendation. Links from email:
- [Connection pooling docs](https://supabase.com/docs/guides/database/connection-management#connections)
- Database settings (add your project ID): https://supabase.com/dashboard/project/[YOUR-PROJECT-ID]/settings/database

Also you can monitor your DB by setting up [Supabase Grafana](https://supabase.com/docs/guides/monitoring-troubleshooting/metrics#deploying-supabase-grafana) locally

**Comment by diffent** [comment]
Thanks, Govind!  Will keep this in mind if we run into it again.  Ours still seemed to be too many calls stacked up before they had a chance to resolve.  Then they timeout'd on the client side before they resolved.  Because you only get "low N" network connections simultaneously on a phone.  8, 10, 12, I forget.  It's not hundreds :)

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.
